File: java/d9142022/BinarySearch.java
package d9142022;

public class BinarySearch {
    public int doSearch(int []arr, int key){
        int left = 0;
        int right = arr.length;
        while (left < right){
            int mid = (left + right) / 2;
            if (arr[mid] > key){
                right = mid - 1;
            }
            else if(arr[mid] < key){
                left = mid + 1;
            }
            else{
                return mid;
            }
        }
        return -1;
    }
}

File: python/2024_04_06/binary_search.py
def bs(arr, k):
    left = 0
    right = len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] < k:
            left = mid + 1
        elif arr[mid] > k:
            right = mid - 1
        else:
            return mid
    return - 1
